prymer/offtarget/offtarget_detector.py

```diff
@@ -191,31 +191,83 @@ def __init__(
+                parameter `max_mismatches_in_three_prime_region` is evaluated. Must be greater than
+                    0.
             max_mismatches_in_three_prime_region: the maximum number of mismatches that are
                 tolerated in the three prime region of each primer defined by
-                three_prime_region_length
+                `three_prime_region_length`. Must be between 0 and `three_prime_region_length`,
+                inclusive.
             max_mismatches: the maximum number of mismatches allowed in the full length primer
-                (including any in the three prime region)
-            max_amplicon_size: the maximum amplicon size to consider amplifiable
+                (including any in the three prime region). Must be greater than or equal to 0.
+            max_amplicon_size: the maximum amplicon size to consider amplifiable. Must be greater
+                than 0.
             cache_results: if True, cache results for faster re-querying
             threads: the number of threads to use when invoking bwa
             keep_spans: if True, [[OffTargetResult]] objects will be reported with amplicon spans
                 populated, otherwise not
             keep_primer_spans: if True, [[OffTargetResult]] objects will be reported with left and
                 right primer spans
             executable: string or Path representation of the `bwa` executable path
+
+        Raises:
+            ValueError: If `max_amplicon_size` is not greater than 0.
+            ValueError: If any of `max_primer_hits`, `max_primer_pair_hits`, or
+                `min_primer_pair_hits` are not greater than or equal to 0.
+            ValueError: If `three_prime_region_length` is not greater than 0.
+            ValueError: If `max_mismatches_in_three_prime_region` is outside the range 0 to
+                `three_prime_region_length`, inclusive.
+            ValueError: If `max_mismatches` is not greater than or equal to 0.
         """
+        errors: list[str] = []
+        if max_amplicon_size < 1:
+            errors.append(f"'max_amplicon_size' must be greater than 0. Saw {max_amplicon_size}")
+        if max_primer_hits < 0:
+            errors.append(
+                f"'max_primer_hits' must be greater than or equal to 0. Saw {max_primer_hits}"
+            )
+        if max_primer_pair_hits < 0:
+            errors.append(
+                "'max_primer_pair_hits' must be greater than or equal to 0. "
+                f"Saw {max_primer_pair_hits}"
+            )
+        if min_primer_pair_hits < 0:
+            errors.append(
+                "'min_primer_pair_hits' must be greater than or equal to 0. "
+                f"Saw {min_primer_pair_hits}"
+            )
+        if three_prime_region_length < 1:
+            errors.append(
+                "'three_prime_region_length' must be greater than 0. "
+                f"Saw {three_prime_region_length}"
+            )
```

Comment:
Given that we use this to drive the "seed region" for bwa I think we should probably enforce a higher miniumum.  @nh13 do you have a thought on what's reasonable?  E.g. 8, 10, 12 or somewhere around there?

Reply:
I agree that since this value is use for the seed length, we should (1) add that explanation to the parameter or class docs, and (2) have a minimum of 8.